### PR TITLE
Apr22 Requisition Tuneup Batch

### PR DESCRIPTION
**Pull request by Kubius**
[FEAT]
## About the PR 

Balance and improvement pass to QM requisitions.

- Bulk of this update is increasing payouts; base payouts and per-item fee modifiers have been buffed, with particular emphasis on Aid contracts due to their time-sensitive nature. A few weightings have also been adjusted.
- A new debug proc, /client/proc/TestMarketReq() ("Requisition Test"), has been introduced. It allows you to directly add a certain path of contract to the market in "test mode", automatically pinning the newly-generated contract (even if it's an Aid contract or a contract is already pinned) for ease of validation. Slightly adjusted the QM console logic to account for this.
- A new Civilian contract has been added, pertaining to construction; flavor-wise, this is earlier in the construction process than the interior outfitting contract, and requires one or two possible resources from a short but varied list.
- The "medical emergency" Aid contract no longer has bandages as a possible requirement, instead requiring styptic powder or silver sulfadiazine delivered within patches. To support this, the reagent entry type has been upgraded to optionally support specification of what the desired reagents must be contained in.
- The "supply chain failure" Aid contract can no longer require medical herbs due to the complexity of acquisition being wildly out of line for a time-limited contract. It also requires less fuel when it's of the fuel variant, and the fuel payout has been drastically increased, receiving over 4x the per-unit reward.
- The "event catering" Civilian contract now requires dramatically fewer of each selected food item. Per-item rewards for food are also some of the more significantly increased fee modifiers.
- The "organ research" Scientific contract can no longer require the brain as a target organ, instead favoring the appendix (which receives a small supplementary bonus due to the appendix's low market valuation, in addition to the blanket buffs from the balance pass).
- The "botanical prototyping" Scientific contract now scales its payout primarily by the amount of unique seed variants that are required, as this is the primary metric of difficulty in fulfilling the contract. Peak payout is slightly lower, but average payout will be dramatically higher.
- Drastically increased the payout from the special-order event which requires particular food items.
- Entries that can use commodity prices are now standardized to not replace an explicitly-specified path if it exists.
- Path checking for stack entries now uses istype, fixing a bug where certain stacks would fail to be accepted by contracts due to being a subtype.
- Return for insufficient goods now actually uses a literal return, preventing erroneous empty sale messages from appearing periodically.
- Item boxes (of the non-storage, "stacking" type) now process through requisition evaluation correctly, by adding real contents to the main list and simulating the un-created contents.
- Items contained in object storage of some sort will now mark that object storage as used by the requisition, instead of the individual contents.
 
## Why's this needed? 

Requisitions are currently somewhat undertuned to serve as an alternative to the more dominant money-making methods for Quartermasters, even taking the "spammy" strategies into account; the new values may be slightly over- or under-tuned in a few cases, but should broadly help ensure that requisitions are worth the time of the people fulfilling them.

(Also fixes up a few wack behaviors.)

## Changelog

```changelog
(u)Kubius
(*)QM requisitions have received a buff / balancing pass, and a new civilian contract variant.
```

### CODE REVIEW

File: code/modules/economy/requisition/rc_aid.dm [hunk 1]
```diff
@@ -106,17 +106,19 @@ ABSTRACT_TYPE(/datum/rc_entry/item/basictool)
 	New()
 		src.name = pick(namevary)
 		src.flavor_desc = "[pick(desc_helpsite)] requires additional supplies [pick(desc_tense)] [pick(desc_crisis)]. [pick(desc_emphasis)]"
-		src.payout += rand(0,40) * 10
+		src.payout += rand(0,60) * 10
+
+		if(prob(60))
+			src.rc_entries += rc_buildentry(/datum/rc_entry/reagent/brutepatch,rand(1,4)*30)
+		if(prob(70))
+			src.rc_entries += rc_buildentry(/datum/rc_entry/reagent/burnpatch,rand(1,4)*30)
```
Comment (frawhst) [inline code/modules/economy/requisition/rc_aid.dm:114]: Will this properly find patches inside patch boxes?

Comment (Kubius) [reply code/modules/economy/requisition/rc_aid.dm:114]: Not currently, as they're not storage items in the conventional sense. Working on a way to account for this.

Comment (Kubius) [reply code/modules/economy/requisition/rc_aid.dm:114]: Now accounted for, including boxes that haven't physically generated their items yet. Not pretty, but functional (tested ingame and should account for any variants of item box).

**Review by frawhst — APPROVED** [review]
seemsgood

**Pull request merged**